lib/matplotlib/sphinxext/plot_directive.py

```diff
@@ -47,6 +47,15 @@
 
 The ``.. plot::`` directive supports the following options:
 
+``:output-base-name:`` : str
+    The base name (without the extension) of the outputted image files. The
+    default is to use the same name as the input script, or the name of
+    the RST document if no script is provided. The string can include the
+    format ``{counter}`` to use an incremented counter. For example,
+    ``'plot-{counter}'`` will create files like ``plot-1.png``, ``plot-2.png``,
+    and so on. If the ``{counter}`` is not provided, two plots with the same
+    output-base-name may overwrite each other.
```

Comment:
Perhaps the extension should actually check for this and just error.

Reply:
OK, nevermind, I undid this change. This seems harder to do than I thought, because you'd need to keep track of which files are generated.